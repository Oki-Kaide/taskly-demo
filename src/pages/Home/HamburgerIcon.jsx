import React from 'react';

const HamburgerIcon = () => (
  <svg width="18px" height="12px" viewBox="0 0 18 12" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlnsXlink="http://www.w3.org/1999/xlink">
    <g id="Page-1" stroke="none" strokeWidth="1" fill="none" fillRule="evenodd">
        <g id="Website-01-(Mobile)" transform="translate(-341.000000, -48.000000)">
            <g id="baseline-menu-24px-(1)" transform="translate(338.000000, 42.000000)">
                <polygon id="Path" points="0 0 24 0 24 24 0 24"></polygon>
                <path d="M4,18 L20,18 C20.5522847,18 21,17.5522847 21,17 C21,16.4477153 20.5522847,16 20,16 L4,16 C3.44771525,16 3,16.4477153 3,17 C3,17.5522847 3.44771525,18 4,18 Z M4,13 L20,13 C20.5522847,13 21,12.5522847 21,12 C21,11.4477153 20.5522847,11 20,11 L4,11 C3.44771525,11 3,11.4477153 3,12 C3,12.5522847 3.44771525,13 4,13 Z M3,7 C3,7.55228475 3.44771525,8 4,8 L20,8 C20.5522847,8 21,7.55228475 21,7 C21,6.44771525 20.5522847,6 20,6 L4,6 C3.44771525,6 3,6.44771525 3,7 Z" id="Shape" fill="#000000"></path>
            </g>
        </g>
    </g>
  </svg>
);

export default HamburgerIcon;
