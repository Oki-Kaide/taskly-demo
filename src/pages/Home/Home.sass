@import '../../styles/variables.sass'
@import '../../styles/mixins.sass'

.main
  display: flex
  flex-direction: column
  justify-content: center
  @include for-phone-only
    min-height: calc(100% - #{$top-header-margin} - #{$top-header-height} - #{$app-store-badge-height})
  @include for-tablet-portrait-up
    min-height: calc(100% - #{$top-header-margin} - #{$top-header-height} - #{$app-store-badge-height})
  @include for-tablet-portrait-landscape-up
    width: 332px
  @include for-tablet-landscape-up
    width: 400px

  h1
    margin-bottom: 50px

  .app-store-badges
    img
      width: 130px

      &:first-of-type 
        margin-right: 10px
    @include for-phone-only
      margin-top: 20px
    @include for-tablet-portrait-up
      margin-top: 120px
    @include for-tablet-portrait-landscape-up
      margin-top: 200px
    @include for-tablet-landscape-up
      position: fixed
      bottom: 40px

