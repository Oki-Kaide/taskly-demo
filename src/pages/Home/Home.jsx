import React from "react";
import PropTypes from "prop-types";
import { RedButton, RedBackground } from "../../components";
import HamburgerIcon from "./HamburgerIcon";
import "./Home.sass";
import '../../App.sass';

const Home = ({openLoginModal, windowWidth, error}) => {
  return (
    <div className="page">
      <div className="page-wrapper">
        <div className="header">
          <img src="./images/taskly-logo.svg" alt="Taskly" />
          <div className="header-buttons">
            {windowWidth > 700 ? (
              <>
                <div className="header-button-item">Services</div>
                <div className="header-button-item">About Us</div>
                <div className="header-button-item">Contact</div>
              </>
            ) : (
              <HamburgerIcon />
            )}
          </div>
        </div>
        <div className="main">
          <h1>Get a virtual assistant. Organize like never before.</h1>
          <RedButton
            text="Connect with Wallet"
            className="login-button"
            handleOnClick={openLoginModal}
          />
          {error ? <h4 className="error">{error}</h4> : null}
          <div className="app-store-badges">
            <img src="./images/google-badge.png" alt="Download on Google" />
            <img src="./images/apple-badge.png" alt="Download on Apple" />
          </div>
        </div>
        <RedBackground side="right" top="20%" right="-20px">
          <img src="./images/peeps.png" alt="Peeps!" />
        </RedBackground>
      </div>
    </div>
  );
};

export default Home;

Home.propTypes = {
  openLoginModal: PropTypes.func.isRequired,
  windowWidth: PropTypes.number.isRequired
}
