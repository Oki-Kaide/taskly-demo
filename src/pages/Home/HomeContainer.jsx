import React from "react";
import Home from "./Home";
import ProtonSDK from "../../utils/proton";

class HomeContainer extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      windowWidth: 0,
    };
  }

  componentDidMount() {
    this.updateWindowWidth();
    window.addEventListener("resize", this.updateWindowWidth);
    /* istanbul ignore next */
    document.addEventListener("backToSelector", () => {
      this.generateLoginRequest();
    });
  }

  updateWindowWidth = () => {
    this.setState({ windowWidth: window.innerWidth });
  };

  /* istanbul ignore next */
  generateLoginRequest = async () => {
    const { setLoggedInState, setErrorState } = this.props;
    const { auth, accountData, error } = await ProtonSDK.login();
    if (error) {
      setErrorState(error);
      return;
    }
    if (!auth || !auth.actor || !auth.permission) {
      return;
    }
      
    setLoggedInState(auth.actor, auth.permission, accountData);
  };

  render() {
    const { windowWidth } = this.state;
    const { error } = this.props;

    return (
      <Home
        openLoginModal={this.generateLoginRequest}
        windowWidth={windowWidth}
        error={error}
      />
    );
  }
}

export default HomeContainer;
