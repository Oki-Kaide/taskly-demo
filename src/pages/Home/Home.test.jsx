import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Home from './Home';
import HamburgerIcon from './HamburgerIcon';

const createTestProps = props => ({
  openLoginModal: jest.fn(),
  windowWidth: 800,
  isLoggingIn: false,
  ...props,
});

describe('Home component', () => {
  let props;

  beforeEach(() => {
    props = createTestProps({});
  });

  it('renders without crashing', () => {
    shallow(<Home {...props} />);
  });

  it('matches the rendered snapshot with default props', () => {
    const component = shallow(<Home {...props} />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('matches the rendered snapshot with isLoggingIn true and smaller window width', () => {
    props = createTestProps({
      windowWidth: 500,
      isLoggingIn: true
    });
    const component = shallow(<Home {...props} />);
    expect(toJson(component)).toMatchSnapshot();
  });
});

describe('HamburgerIcon component', () => {
  it('renders without crashing', () => {
    shallow(<HamburgerIcon />);
  });

  it('matches the rendered snapshot with default props', () => {
    const component = shallow(<HamburgerIcon />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
