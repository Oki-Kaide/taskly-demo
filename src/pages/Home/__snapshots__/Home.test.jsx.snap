// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HamburgerIcon component matches the rendered snapshot with default props 1`] = `
<svg
  height="12px"
  version="1.1"
  viewBox="0 0 18 12"
  width="18px"
  xmlns="http://www.w3.org/2000/svg"
  xmlnsXlink="http://www.w3.org/1999/xlink"
>
  <g
    fill="none"
    fillRule="evenodd"
    id="Page-1"
    stroke="none"
    strokeWidth="1"
  >
    <g
      id="Website-01-(Mobile)"
      transform="translate(-341.000000, -48.000000)"
    >
      <g
        id="baseline-menu-24px-(1)"
        transform="translate(338.000000, 42.000000)"
      >
        <polygon
          id="Path"
          points="0 0 24 0 24 24 0 24"
        />
        <path
          d="M4,18 L20,18 C20.5522847,18 21,17.5522847 21,17 C21,16.4477153 20.5522847,16 20,16 L4,16 C3.44771525,16 3,16.4477153 3,17 C3,17.5522847 3.44771525,18 4,18 Z M4,13 L20,13 C20.5522847,13 21,12.5522847 21,12 C21,11.4477153 20.5522847,11 20,11 L4,11 C3.44771525,11 3,11.4477153 3,12 C3,12.5522847 3.44771525,13 4,13 Z M3,7 C3,7.55228475 3.44771525,8 4,8 L20,8 C20.5522847,8 21,7.55228475 21,7 C21,6.44771525 20.5522847,6 20,6 L4,6 C3.44771525,6 3,6.44771525 3,7 Z"
          fill="#000000"
          id="Shape"
        />
      </g>
    </g>
  </g>
</svg>
`;

exports[`Home component matches the rendered snapshot with default props 1`] = `
<div
  className="page"
>
  <div
    className="page-wrapper"
  >
    <div
      className="header"
    >
      <img
        alt="Taskly"
        src="./images/taskly-logo.svg"
      />
      <div
        className="header-buttons"
      >
        <div
          className="header-button-item"
        >
          Services
        </div>
        <div
          className="header-button-item"
        >
          About Us
        </div>
        <div
          className="header-button-item"
        >
          Contact
        </div>
      </div>
    </div>
    <div
      className="main"
    >
      <h1>
        Get a virtual assistant. Organize like never before.
      </h1>
      <RedButton
        className="login-button"
        handleOnClick={[MockFunction]}
        isLoading={false}
        text="Connect with Wallet"
      />
      <div
        className="app-store-badges"
      >
        <img
          alt="Download on Google"
          src="./images/google-badge.png"
        />
        <img
          alt="Download on Apple"
          src="./images/apple-badge.png"
        />
      </div>
    </div>
    <RedBackground
      bottom="auto"
      left="auto"
      right="-20px"
      side="right"
      top="20%"
    >
      <img
        alt="Peeps!"
        src="./images/peeps.png"
      />
    </RedBackground>
  </div>
</div>
`;

exports[`Home component matches the rendered snapshot with isLoggingIn true and smaller window width 1`] = `
<div
  className="page"
>
  <div
    className="page-wrapper"
  >
    <div
      className="header"
    >
      <img
        alt="Taskly"
        src="./images/taskly-logo.svg"
      />
      <div
        className="header-buttons"
      >
        <HamburgerIcon />
      </div>
    </div>
    <div
      className="main"
    >
      <h1>
        Get a virtual assistant. Organize like never before.
      </h1>
      <RedButton
        className="login-button"
        handleOnClick={[MockFunction]}
        isLoading={false}
        text="Connect with Wallet"
      />
      <div
        className="app-store-badges"
      >
        <img
          alt="Download on Google"
          src="./images/google-badge.png"
        />
        <img
          alt="Download on Apple"
          src="./images/apple-badge.png"
        />
      </div>
    </div>
    <RedBackground
      bottom="auto"
      left="auto"
      right="-20px"
      side="right"
      top="20%"
    >
      <img
        alt="Peeps!"
        src="./images/peeps.png"
      />
    </RedBackground>
  </div>
</div>
`;
